src/H5FDonion.c

```diff
@@ -351,6 +354,65 @@ H5Pset_fapl_onion(hid_t fapl_id, const H5FD_onion_fapl_info_t *fa)
     FUNC_LEAVE_API(ret_value)
 } /* end H5Pset_fapl_onion() */
 
+hssize_t
+H5FDget_onion_revision_count(const char *filename, hid_t fapl_id)
+{
+    H5P_genplist_t *plist        = NULL;
+    ssize_t         ret_value    = 0;
+    H5FD_t *        file_drv_ptr = NULL;
+    FUNC_ENTER_API(FAIL)
+
+    /* Make sure using the correct driver */
+    if (NULL == (plist = H5P_object_verify(fapl_id, H5P_FILE_ACCESS)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "not a valid FAPL ID")
+
+    if (H5FD_ONION != H5P_peek_driver(plist))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "not a Onion VFL driver")
+
+    /* Open the file with the driver */
+    if (NULL == (file_drv_ptr = H5FD_open(filename, H5F_ACC_RDONLY, fapl_id, HADDR_UNDEF)))
+        HGOTO_ERROR(H5E_VFL, H5E_CANTOPENFILE, FAIL, "unable to open file with onion driver")
+
+    /* Call the private function */
+    if ((ret_value = H5FD__get_onion_revision_count(file_drv_ptr)) < 0)
+        HGOTO_ERROR(H5E_VFL, H5E_CANTGET, FAIL, "failed to get the number of revisions")
+
+    /* Close H5FD_t structure pointer */
+    if (H5FD_close(file_drv_ptr) < 0)
+        HGOTO_ERROR(H5E_VFL, H5E_CANTCLOSEFILE, FAIL, "unable to close file")
+
+done:
+    FUNC_LEAVE_API(ret_value)
+}
+
+static hssize_t
+H5FD__get_onion_revision_count(H5FD_t *file)
+{
+    hssize_t  ret_value = 0;
+    uint64_t  op_code;
+    uint64_t  flags;
+    uint64_t *num_revisions = NULL;
+
+    FUNC_ENTER_PACKAGE
+
+    if (NULL == (num_revisions = HDmalloc(sizeof(uint64_t))))
```

Comment:
Normally we HDassert on all pointers on function entry

Reply:
Inside the library, we use H5MM_malloc instead of HD. I also prefer calloc to initialize memory.

Reply:
> Normally we HDassert on all pointers on function entry

In the past, I was told to use the error report stack instead of assertion.  Has it changed?